src/App.jsx

```diff
@@ -14,57 +16,66 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
+  const clearSelection = () => setSelectedGood('');
+  const handleSelection = (good, isSelected) =>
+    isSelected ? clearSelection() : setSelectedGood(good);
```

Comment:
here you can update state based on the previous state 
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state

```suggestion
  const handleSelection = (good) =>
   setSelectedGood(previousGood => (previousGood === good ? '' : good));
```